realm/src/main/java/io/realm/BaseRealm.java

```diff
@@ -176,9 +177,13 @@ public void removeAllChangeListeners() {
         changeListeners.clear();
     }
 
-    protected void removeHandler(Handler handler) {
-        handler.removeCallbacksAndMessages(null);
+    /**
+     * Remove and stop the current thread handler as gracefully as possible.
+     */
+    protected void removeHandler() {
         handlers.remove(handler);
+        // Warning: This only clears the Looper queue. Handler.Callback is not removed.
+        handler.removeCallbacksAndMessages(null);
```

Comment:
This will only remove any pending posts, what about preventing other threads from posting?